roles/host-provision/tasks/proxy.yml

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 Google LLC
```

Comment:
These tasks all depend on a HTTP proxy being configured on the control node.  Can we ensure that it's in place? 

And possibly bigger questions  around how to configure access to external services (primarily Red Hat license servers and software updates, I imagine?)

Reply:
This is a good catch. Definitely the whole `proxy.yml` is predicated on the fact that there will be proxy server running in the control-node/cloudVM.

My initial thought was to add conditionals in the tasks in `proxy.yml` that looks for a output similar to the following:
```bash
✔ ~/050521-mntrl-instlln/bms-toolkit [host-provision ↓·2|✚ 2…19] 
18:22 $ service squid status
Redirecting to /bin/systemctl status squid.service
● squid.service - Squid caching proxy
   Loaded: loaded (/usr/lib/systemd/system/squid.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2021-08-20 09:34:13 PDT; 1 months 22 days ago
 Main PID: 27941 (squid)
   CGroup: /system.slice/squid.service
           ├─27941 /usr/sbin/squid -f /etc/squid/squid.conf
           ├─27943 (squid-1) -f /etc/squid/squid.conf
           └─27949 (logfile-daemon) /var/log/squid/access.log

Aug 20 09:34:13 control-node systemd[1]: Starting Squid caching proxy...
Aug 20 09:34:13 control-node squid[27941]: Squid Parent: will start 1 kids
Aug 20 09:34:13 control-node squid[27941]: Squid Parent: (squid-1) process 27943 started
Aug 20 09:34:13 control-node systemd[1]: Started Squid caching proxy.
```

Then realized that one of the very first tasks we have in our `proxy.yml` is indeed checking for 200:
```yaml
- name: proxy | Validate proxy connectivity
  uri:
    url: https://www.google.com
    status_code: 200
  environment:
    https_proxy: "http://{{ proxy_ip }}:3128"
  tags: host-provision
```

The above causes the script to exit and not execute the other tasks that follow if the control-node is not indeed acting as a proxy server. This check probably should suffice.

Since this PR is focused on the BMS hosts, installing a proxy server, like Squid Proxy, on the control-node can be kept to a separate code change.

Will add in the comments section about the link (https://cloud.google.com/vpc/docs/special-configurations#proxyvm) you provided in another comment.

Reply:
As discussed, removing the proxy/internet connectivity check from here, as the message itself is self-sufficient:
```
[root@BMS-HOST-1 yum.repos.d]# /sbin/subscription-manager register --force --auto-attach --username VALUE_SPECIFIED_IN_NO_LOG_PARAMETER --password asdf
Unregistering from: subscription.rhsm.redhat.com:443/subscription
Network error. Please check the connection details, or see /var/log/rhsm/rhsm.log for more information.
```